swarm/network/discovery_test.go

```diff
@@ -171,7 +174,9 @@ func testInitialPeersMsg(t *testing.T, peerPO, peerDepth int) {
 	}
 
 	// pivotDepth is the advertised depth of the pivot node we expect in the outgoing subPeersMsg
+	hive.Kademlia.lock.RLock() // protect Kademlia.conns that are read in Kademlia.saturation()
```

Comment:
Similar as above, we are talking to strangers => leaking abstractions. (Law of Demeter)

Here, what I would consider is adding `hive.Saturation()`. That would be the locking version of `hive.saturation()` What do you think?

Reply:
Same as above. Thanks. I will add the new method.